tests/Resources/ti.ui.animation.test.js

```diff
@@ -0,0 +1,420 @@
+
+		describe('.height', () => {
+			it('is a Number', () => {
+				instance.height = 100;
+				should(instance).have.a.property('height').which.is.a.Number();
+			});
+		});
+
+		describe('.left', () => {
+			it('is a Number', () => {
+				instance.left = 0;
+				should(instance).have.a.property('left').which.is.a.Number();
+			});
+		});
+
+		describe('.opacity', () => {
+			it('is a Number', () => {
+				instance.opacity = 0.5;
+				should(instance).have.a.property('opacity').which.is.a.Number();
+			});
+		});
+
+		describe.ios('.opaque', () => {
+			it('is a Boolean', () => {
+				instance.opaque = false;
+				should(instance).have.a.property('opaque').which.is.a.Boolean();
+			});
+		});
+
+		describe('.repeat', () => {
+			it('is a Number', () => {
+				should(instance).have.a.property('repeat').which.is.a.Number();
+			});
+
+			it('defaults to 1 (no repeat)', () => {
+				should(instance.repeat).eql(1);
+			});
+		});
+
+		describe('.right', () => {
+			it('is a Number', () => {
+				instance.right = 20;
+				should(instance).have.a.property('right').which.is.a.Number();
+			});
+		});
+
+		describe.ios('.springVelocity', () => {
+			it('is a Number', () => {
+				instance.springVelocity = 0.5;
+				should(instance).have.a.property('springVelocity').which.is.a.Number();
+			});
+		});
+
+		describe('.top', () => {
+			it('is a Number', () => {
+				instance.top = 40;
+				should(instance).have.a.property('top').which.is.a.Number();
+			});
+		});
+
+		describe('.transform', () => {
+			it('is a Titanium.UI.Matrix2D,Titanium.UI.Matrix3D', () => {
+				instance.transform = Ti.UI.createMatrix2D({
+					scale: 2
+				});
+				should(instance).have.a.property('transform').which.is.an.Object();
+				const matrixApiName = Ti.Platform.osname === 'android' ? 'Ti.UI.Matrix2D' : 'Ti.UI.2DMatrix';
+				should(instance.transform.apiName).eql(matrixApiName);
+			});
+		});
+
+		describe.ios('.transition', () => {
+			it('is a Number', () => {
+				instance.transition = Ti.UI.iOS.AnimationStyle.CROSS_DISSOLVE;
+				should(instance).have.a.property('transition').which.is.a.Number();
+			});
+
+			it('is one of Ti.UI.iOS.AnimationStyle.CURL_*', () => {
+				instance.transition = Ti.UI.iOS.AnimationStyle.FLIP_FROM_TOP;
+				should([
+					Ti.UI.iOS.AnimationStyle.CURL_DOWN,
+					Ti.UI.iOS.AnimationStyle.CURL_UP,
+					Ti.UI.iOS.AnimationStyle.FLIP_FROM_LEFT,
+					Ti.UI.iOS.AnimationStyle.FLIP_FROM_RIGHT,
+					Ti.UI.iOS.AnimationStyle.FLIP_FROM_TOP,
+					Ti.UI.iOS.AnimationStyle.FLIP_FROM_BOTTOM,
+					Ti.UI.iOS.AnimationStyle.CROSS_DISSOLVE,
+					Ti.UI.iOS.AnimationStyle.NONE,
+				]).containEql(instance.transition);
+			});
+		});
+
+		describe.ios('.view', () => {
+			it('is a Titanium.UI.View', () => {
+				instance.view = Ti.UI.createView();
+				should(instance).have.a.property('view').which.is.an.Object();
+				should(instance.view).have.a.property('apiName').which.is.eql('Ti.UI.View');
+			});
+		});
+
+		describe.ios('.visible', () => {
```

Comment:
another property which feels like it should have a known.documented default value - but perhaps it depends on the view?